docs/architecture/adr-009-evidence-module.md

```diff
@@ -0,0 +1,205 @@
+First, we define the `Evidence` interface type. The `x/evidence` module may implement
+its own types that can be used by many chains (e.g. `CounterFactualEvidence`).
+In addition, other modules may implement their own `Evidence` types in a similar
+manner in which governance is extensible. It is important to note any concrete 
+type implementing the `Evidence` interface may include arbitrary fields such as
+an infraction time. We want the `Evidence` type to remain as flexible
+as possible.
+
+However, when submitting evidence to the `x/evidence` module, it must be submitted
+as an `Infraction` which includes mandatory fields outlined below.
+ 
+```go
+type Evidence interface {
+	Jailable() bool
+	Route() string
+	Type() string
+	ValidateBasic() sdk.Error
+	String() string
+}
+
+type Infraction struct {
+	Evidence
+	
+	ConsensusAddress    sdk.ConsAddress
+	InfractionHeight    int64
+	Power               int64
+}
+```
+
+### Routing & Handling
+
+Each `Evidence` type must map to a specific unique route and be registered with
+the `x/evidence` module. It accomplishes this through the `Router` implementation. 
+
+```go
+type Router interface {
+	AddRoute(r string, h Handler) (rtr Router)
+	HasRoute(r string) bool
+	GetRoute(path string) (h Handler)
+	Seal()
+}
+```
+
+Upon successful routing through the `x/evidence` module, the `Evidence` type
+is passed through a `Handler`. This `Handler` is responsible for executing all
+corresponding business logic necessary for verifying the evidence. If no error
+is returned, the `Evidence` is considered valid.
+
+```go
+type Handler func(ctx sdk.Context, evidence Evidence) sdk.Error
+```
+
+### Submission
+
+Assuming the `Evidence` is valid, the corresponding slashing penalty is invoked
+for the `Evidence`'s `Type`. Keep in mind the slashing penalty for any `Type` can
+be configured through governance.
+
+```go
+type MsgSubmitInfraction struct {
+	Infraction
+}
+
+func handleMsgSubmitInfraction(ctx sdk.Context, keeper Keeper, msg MsgSubmitEvidence) sdk.Result {
+	if err := keeper.SubmitInfraction(ctx, msg.Infraction); err != nil {
+		return err.Result()
+	}
+    
+	// emit events...
+
+	return sdk.Result{ 
+		// ...
+	}
+}
+```
+
+The `x/evidence` module's keeper is responsible for matching the `Evidence` against
+the module's router. Upon success the validator is slashed and the infraction is
+persisted. In addition, the validator is jailed is the `Evidence` type is configured
+to do so. 
+
+```go
+func (k Keeper) SubmitInfraction(ctx sdk.Context, infraction Infraction) sdk.Error {
+	handler := keeper.router.GetRoute(infraction.Evidence.Route())
+	if err := handler(cacheCtx, infraction.Evidence); err != nil {
+		return ErrInvalidEvidence(keeper.codespace, err.Result().Log)
+	}
+	
+	keeper.stakingKeeper.Slash(
+		ctx,
+		infraction.ConsensusAddress, 
+		infraction.InfractionHeight, 
+		infraction.Power, 
+		keeper.GetSlashingPenalty(ctx, infraction.Evidence.Type()),
+	)
+	
+	if infraction.Evidence.Jailable() {
```

Comment:
Why this is a method on the interface instead of a parameter (like the penalty is)?

Reply:
Updated to be part of params now; good idea.